### ISSUE TITLE
question about input of spconv_unet_v1m3_pdnorm

### ISSUE THREAD

**Issue by qiuzidian** [OpenGVLab/PonderV2#26]
Can you tell me the meaning and shape of each tensor in input_dict in spconv_unet_v1m3_pdnorm.py?
```
    def forward(self, input_dict):
        grid_coord = input_dict["grid_coord"]
        feat = input_dict["feat"]
        offset = input_dict["offset"]
        condition = input_dict["condition"][0]
        context = input_dict["context"] if "context" in input_dict.keys() else None
```
Is it compatible with the following inputs?
```
data = np.load("examples/input.npz")
feat, xyz, batch = data["feat"], data["xyz"], data["batch"]
# feats: [N, 6], RGB, Normal
# xyz: [N, 3],
# batch: [N],
```

**Comment by HaoyiZhu** [comment]
Hi, 

- `grid_coord` is the int coordinate after [GridSample](https://github.com/OpenGVLab/PonderV2/blob/main/ponder/datasets/transform.py#L1079). It means the index or coordinate on the voxelized space.
- `feat` is usually rgb + normal
- `offset` is the separator of point clouds in batch data. A visual illustration of batch and offset is as follows:
<p align="center">
  <img src="https://github.com/user-attachments/assets/74f51677-7b36-4bc7-88cf-324bf777d46a" alt="Image" width="300"/>
</p>

- `condition` is used to identify the dataset source which can be chosen from "Structured3D", "ScanNet", "S3DIS".
- different `condition` has different `context` values, you can see from [here](https://github.com/OpenGVLab/PonderV2/blob/772fe3a0dad37e0cb9f604bedb996c799ccd688e/ponder/models/ponder/ponder_indoor_base.py#L164-L172). It is PPT-related and is used to give different batch norm parameters.

**Comment by qiuzidian** [comment]
Thank you, I understand now!

**Comment by qiuzidian** [comment]
Sorry to bother you, but I have another question. Is the context necessary? I just want to use pretrained SparseUNet as the feature extractor for my point cloud. What if my point cloud comes from other datasets?

**Comment by HaoyiZhu** [comment]
You can randomly choose one from the three.

**Comment by qiuzidian** [comment]
thank you!
